Отправка сообщений в Telegram

Alginin opened the pull request:
2. Создан бот.
3. Создана вся необходимая конфигурация для коммуникации в application.yaml.
4. TelegramService реализует интерфейс NotificationService.
5. Написаны unit-тесты.

Alginin committed:
1. Добавлены зависимости на Telegram SDK.
2. Создан бот.
3. Создана вся необходимая конфигурация для коммуникации в application.yaml.
4. TelegramService реализует интерфейс NotificationService.
5. Написаны unit-тесты.

Peregruzochka commented on src/main/resources/application.yaml:
```diff
@@ -25,6 +25,10 @@ spring:
   messages:
     basename: messages
 
+  telegram-bot:
```
вынеси эту инфу в отдельный yaml
посмотри, как добавлять новый yaml к application.yaml

Peregruzochka commented on src/main/java/faang/school/notificationservice/config/telegram/BotConfig.java:
```diff
@@ -0,0 +1,14 @@
+
+@Configuration
+@Getter
+public class BotConfig {
```
Лучше чтобы этот класс создавал `bean TelegramBot` и принимал в себя параметры name и токен
Соответственно у TelegramBot убрать аннотацию @Component

Peregruzochka commented on src/main/java/faang/school/notificationservice/telegram/TelegramBot.java:
```diff
@@ -0,0 +1,29 @@
+
+    @Override
+    public void onUpdateReceived(Update update) {
+
```
Прикол в том, что тыв не можешь отправить сообщение пользователю не зная его telegramId (чтобы его узнать, необходимо, чтобы пользователь как минимум зашел в бота) таким образом стоит подумать как собирать и где хранить эти telegramId

Но возможно это задача другого микросервиса/сервиса, и тебе не надо сейчас париться на этот счет

Peregruzochka commented on src/main/java/faang/school/notificationservice/service/TelegramBotService.java:
```diff
@@ -0,0 +1,40 @@
+        if (user.getPreference() != UserDto.PreferredContact.TELEGRAM || user.getTelegramId() == null) {
+            log.info("It is not possible to send a notification via Telegram to the user {} {}", user.getId(), user.getUsername());
+            return;
+        }
```
Валидацию можно в отдельный приватный метод
log.error в данном случае а не log.info

Peregruzochka reviewed (commented):
вери гуд

Alginin committed:
Выполнил работу над кросс-ревью

mrkorolev commented on src/main/java/faang/school/notificationservice/service/TelegramBotService.java:
```diff
@@ -0,0 +1,44 @@
+
+    @Override
+    public void send(UserDto user, String message) {
+        validation(user, message);
```
Нейминг - можно и получше придумать

mrkorolev reviewed (commented):
Мне нравится )

20cognosce commented on src/main/resources/telegram-bot.yaml:
```diff
@@ -0,0 +1,3 @@
+telegram-bot:
```
Что это за yaml
сделай конфигурационный файл application-telegram.yaml и включи его в основной
через include

Его название просто не сочетается ни с чем

20cognosce commented on .gitignore:
```diff
@@ -53,3 +53,6 @@ out/
 .vscode/
+
+### Telegram Bot ###
+telegram-bot.yaml
```
Может тогда это в env вынести, если ты не хочешь его версионировать
Почему он тогда попал на ПР, интересно

Определись уж, добавлять его в гит или нет

20cognosce commented on src/main/java/faang/school/notificationservice/telegram/TelegramBot.java:
```diff
@@ -0,0 +1,29 @@
+    private final BotConfig botConfig;
+
+    @Override
+    public void onUpdateReceived(Update update) {
```
Почему пустой метод, что он делает?

20cognosce commented on src/main/java/faang/school/notificationservice/service/TelegramBotService.java:
```diff
@@ -0,0 +1,44 @@
+                .build();
+
+        try {
+            telegramBot.execute(sendMessage);
```
А кому это сообщение ушло? Боту?
Сделай sequence диаграмму и проведи демо работы твоей фичи, подготовь к дейли пожалуйста

Alginin committed:
Частичное иправление ревью

Alginin committed:
Внесенные изменения:
1. Регистрация бота в TelegramBotsApi
2. Пустой метод onUpdateReceived теперь выполняет функцию "эхо-бота"
